server/src/test/java/org/opensearch/search/approximate/ApproximatePointRangeQueryTests.java

```diff
@@ -86,301 +293,290 @@ public void testApproximateRangeWithDefaultSize() throws IOException {
+                    assertEquals(topDocs.totalHits, new TotalHits(1000, TotalHits.Relation.EQUAL_TO));
                 }
             }
         }
     }
 
-    public void testApproximateRangeWithSizeDefault() throws IOException {
+    public void testApproximateRangeShortCircuitAscSort() throws IOException {
         try (Directory directory = newDirectory()) {
             try (RandomIndexWriter iw = new RandomIndexWriter(random(), directory, new WhitespaceAnalyzer())) {
                 int dims = 1;
-
-                long[] scratch = new long[dims];
                 int numPoints = 1000;
                 for (int i = 0; i < numPoints; i++) {
                     Document doc = new Document();
-                    for (int v = 0; v < dims; v++) {
-                        scratch[v] = i;
-                    }
-                    doc.add(new LongPoint("point", scratch));
+                    numericType.addField(doc, numericType.fieldName, i);
+                    numericType.addDocValuesField(doc, numericType.fieldName, i);
                     iw.addDocument(doc);
                 }
                 iw.flush();
                 iw.forceMerge(1);
-                final int size = 10;
                 try (IndexReader reader = iw.getReader()) {
-                    try {
-                        long lower = 0;
-                        long upper = 45;
-                        Query approximateQuery = new ApproximatePointRangeQuery(
-                            "point",
-                            pack(lower).bytes,
-                            pack(upper).bytes,
-                            dims,
-                            size,
-                            null,
-                            ApproximatePointRangeQuery.LONG_FORMAT
-                        );
-                        IndexSearcher searcher = new IndexSearcher(reader);
-                        TopDocs topDocs = searcher.search(approximateQuery, size);
-                        assertEquals(size, topDocs.scoreDocs.length);
-                    } catch (IOException e) {
-                        throw new RuntimeException(e);
+                    long lower = 0;
+                    long upper = 20;
+                    final int size = 10;
+                    Query approximateQuery = new ApproximatePointRangeQuery(
+                        numericType.fieldName,
+                        numericType.encode(lower),
+                        numericType.encode(upper),
+                        dims,
+                        size,
+                        SortOrder.ASC,
+                        numericType.format
+                    );
+                    Query exactQuery = numericType.rangeQuery(numericType.fieldName, lower, upper);
+                    IndexSearcher searcher = new IndexSearcher(reader);
+                    Sort sort;
+                    if (numericType == NumericType.HALF_FLOAT || numericType == NumericType.UNSIGNED_LONG) {
```

Comment:
Same here get from NumericType enum. 

Idea is while writing tests to add support for a new type should just require adding a new implementation of NumericType. The test framework should then run for every type and only depend on NumericType for specific things